template_scripts/fw_creel.Rmd

```diff
@@ -813,76 +715,53 @@ map_df(estimates_bss, ~.x$overview) |>
-#   )
-# }
+```{r transform_estimates, eval = FALSE}
+#Extract and transform model estimates into a common format
 
 #Export data from one or both models
 if (params$export %in% c("database", "local") && !params$model_used %in% c("PE only", "BSS only", "Both models" )) {
-  cat("Invalid value for model_used parameter. Must select 'PE only', 'BSS only', or 'Both models'.")
+  cat("\nInvalid value for export or model_used parameter.")
 } else {
-
-  # Initialize objects for storing processed data
-  # also serves as placeholder for other model if only one was used
-  transformed_pe_data <- list(
-    pe_effort = data.frame(),
-    pe_catch = data.frame(),
-    pe_stratum_effort = data.frame(),
-    pe_stratum_catch = data.frame(),
-    pe_summarized_effort = data.frame(),
-    pe_summarized_catch = data.frame()
-  )
   
-  transformed_bss_data <- list(
-    bss_stratum = data.frame(),
-    bss_summarized = data.frame(),
-    bss_stratum_catch = data.frame(),
-    bss_stratum_effort = data.frame(),
-    bss_summarized_catch = data.frame(),
-    bss_summarized_effort = data.frame()
-  )
-  
-  if (params$model_used == "PE only") {
-    export_estimates(params, estimates_pe, estimates_bss = NULL)
+  #Check that at least one model output is provided
+  if (exists("estimates_pe") == "FALSE" && exists("estimates_bss") == "FALSE") {
+    stop("At least one of 'estimates_pe' or 'estimates_bss' must be supplied.")
   }
   
-  if (params$model_used == "BSS only") {
-    export_estimates(params, estimates_pe = NULL, estimates_bss)
-  }
+  # Process PE estimates to common format
+  if (exists("estimates_pe") == "TRUE" && params$model_used == "PE only") {
+    transformed_pe_data <- process_estimates_pe(analysis_lut, estimates_pe)
+  } else {print("PE model not used.")}
   
-  if (params$model_used == "Both models") {
-    export_estimates(params, estimates_pe, estimates_bss)
-  }
+  # Process BSS estimates to common format
+  if (exists("estimates_bss") == "TRUE" && params$model_used == "BSS only") {
+    transformed_bss_data <- process_estimates_bss(params, analysis_lut, estimates_bss)
+  } else {print("BSS model not used.")}
+  
+  # Process PE and BSS model estimates to common format
+  if (exists("estimates_pe") == "TRUE" && exists("estimates_bss") == "TRUE" && params$model_used == "Both models") {
+    transformed_pe_data <- process_estimates_pe(analysis_lut, estimates_pe)
+    transformed_bss_data <- process_estimates_bss(params, analysis_lut, estimates_bss)
+  } else {print("Only one model transformed as one model output does not exist or params$model_used does not equal `Both models`.")}
+  
+  #add parameter data_grade as a field on all tables
+  map_data_grade(params, transformed_bss_data, transformed_pe_data)
+  
+  #Combine PE and BSS and perform other transformation steps
+  creel_estimates <- transform_estimates(dwg, transformed_pe_data, transformed_bss_data)  
```

Comment:
Similar issue to above comment, user has to manually assign transformed_pe_data = NULL for this to work in the BSS_only use case. @ctholley

Reply:
bcf05064286c2ea010f9e576d902839ffb252400 should cover this issue as well. I tested a "BSS only" with no existing estimates_pe or transformed_pe_data objects and it processed BSS only as expected.